### PR TITLE
Integration tests infrastructure refactoring. Separation of creation …

### PR DESCRIPTION
**Pull request by alexeykiselev**
…of nodes containers.

Functional options implemented for itests configuration.

### CODE REVIEW

File: itests/docker/docker.go
Comment (alexeykiselev) [reply itests/docker/docker.go:141]: Don't know why. Because of levels? It's for integration tests, nobody configure levels here. But slog text messages format not so nice because they try to keep the structure.

Comment (nickeskov) [reply itests/docker/docker.go:143]: Okay, fair enough

**Review by nickeskov — DISMISSED** [review]

**Review by nickeskov — APPROVED** [review]

File: itests/docker/docker.go
Comment (nickeskov) [inline itests/docker/docker.go:234]: Maybe make a `const`?

**Review by nickeskov — CHANGES_REQUESTED** [review]

File: itests/clients/node_client.go
Comment (nickeskov) [reply itests/clients/node_client.go:61]: Done

File: itests/clients/node_client.go
Comment (nickeskov) [reply itests/clients/node_client.go:141]: Done

File: itests/clients/node_client.go
Comment (nickeskov) [reply itests/clients/node_client.go:117]: Done

File: itests/clients/node_client.go
Comment (nickeskov) [reply itests/clients/node_client.go]: Done

File: itests/clients/node_client.go
Comment (nickeskov) [reply itests/clients/node_client.go]: Done, removed

File: itests/clients/node_client.go
Comment (nickeskov) [reply itests/clients/node_client.go]: Fixed

File: itests/docker/docker.go
Comment (alexeykiselev) [reply itests/docker/docker.go:234]: Fixed

File: itests/clients/node_client.go
Comment (alexeykiselev) [reply itests/clients/node_client.go:61]: There is a timeout inside the `GetHeight` call, but I think you're talking about global timeout on both calls. I've added it.

File: itests/clients/universal_client.go
Comment (alexeykiselev) [reply itests/clients/universal_client.go:24]: Removed

File: itests/clients/universal_client.go
Comment (alexeykiselev) [reply itests/clients/universal_client.go:32]: no need to fix, function removed

File: itests/clients/node_client.go
Comment (nickeskov) [inline itests/clients/node_client.go:90]: please, add error check

Comment (alexeykiselev) [reply itests/clients/node_client.go]: Fixed

**Comment by nickeskov** [comment]
LGTM, will approve after #1505 will be merged to master.

**Review by nickeskov — APPROVED** [review]

**Pull request merged**
